diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1336 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+ !> @brief This code will be used by the preprecessor to generate an implementation
+ !! to the module procudure for the fieldbuff_update interface. The
+  !! generated function is a wrapper calling 4D field/5D buffer version of the same.
+FUNCTION FMS_DIAG_FBU_3D_PNAME_(ofield_cfg, ofield_index_cfg, field_d, sample, &
+  & ofb, ofc, bbounds, count_0d, num_elements, mask, weight1, missvalue,  &
+  & field_num_threads, field_active_omp_level, issued_mask_ignore_warning, &
+  & l_start, l_end, err_msg,  err_msg_local ) result( succeded )
+    TYPE(fms_diag_outfield_type), INTENT(in):: ofield_cfg  !< The ofield_cfg object
```

Comment:
These type names all need to be updated to the correct format.

Reply:
Done